homeassistant/components/openuv/__init__.py

```diff
@@ -82,33 +160,76 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
 
+    @callback
+    def extract_openuv(func: Callable) -> Callable:
+        """Define a decorator to get the correct OpenUV object for a service call."""
+
+        async def wrapper(call: ServiceCall) -> None:
+            """Wrap the service function."""
+            openuv = async_get_openuv_for_service_call(hass, call)
+
+            try:
+                await func(call, openuv)
+            except OpenUvError as err:
+                raise HomeAssistantError(
+                    f'Error while executing "{call.service}": {err}'
+                ) from err
+
+        return wrapper
+
     @_verify_domain_control
-    async def update_data(_: ServiceCall) -> None:
+    @extract_openuv
+    async def update_data(call: ServiceCall, openuv: OpenUV) -> None:
         """Refresh all OpenUV data."""
         LOGGER.debug("Refreshing all OpenUV data")
+        async_log_deprecated_service_call(
+            hass,
+            call,
+            "homeassistant.update_entity",
+            "binary_sensor.protection_window, sensor.current_uv_index",
+            "2022.12.0",
+        )
         await openuv.async_update()
         async_dispatcher_send(hass, TOPIC_UPDATE)
 
     @_verify_domain_control
-    async def update_uv_index_data(_: ServiceCall) -> None:
+    @extract_openuv
+    async def update_uv_index_data(call: ServiceCall, openuv: OpenUV) -> None:
         """Refresh OpenUV UV index data."""
         LOGGER.debug("Refreshing OpenUV UV index data")
+        async_log_deprecated_service_call(
+            hass,
+            call,
+            "homeassistant.update_entity",
+            "sensor.current_uv_index",
+            "2022.12.0",
+        )
         await openuv.async_update_uv_index_data()
         async_dispatcher_send(hass, TOPIC_UPDATE)
 
     @_verify_domain_control
-    async def update_protection_data(_: ServiceCall) -> None:
+    @extract_openuv
+    async def update_protection_data(call: ServiceCall, openuv: OpenUV) -> None:
         """Refresh OpenUV protection window data."""
         LOGGER.debug("Refreshing OpenUV protection window data")
+        async_log_deprecated_service_call(
+            hass,
+            call,
+            "homeassistant.update_entity",
+            "binary_sensor.protection_window",
```

Comment:
Is this the correct entity id if they have multiple?

Reply:
10-4, see above.

Reply:
Latest shows same result as above with this Repair entry:

![CleanShot 2022-09-17 at 16 46 32](https://user-images.githubusercontent.com/47216/190878813-c14a8da5-76f5-48c8-b57a-9fe50a2c4469.png)

![CleanShot 2022-09-17 at 17 04 35](https://user-images.githubusercontent.com/47216/190879082-0ea3097c-91c9-4339-aeb9-bd8d5cae5c65.png)